kong/db/declarative/init.lua

```diff
@@ -926,7 +926,43 @@ do
     local default_ws
     ok, err, default_ws = declarative.load_into_cache(entities, meta, hash, SHADOW)
     if ok then
-      ok, err = worker_events.post("declarative", "flip_config", default_ws)
+      local router_hash
+      local plugins_hash
+      local balancer_hash
+      if hashes then
+        if hashes.services ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+          if hashes.routes ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+            router_hash = md5(hashes.services .. hashes.routes)
+          else
+            router_hash = hashes.services
```

Comment:
if services is nont empty, and routes is empty, should we keep router_has being EMPTY_CONFIG_HASH? as we don't support service-less routes now.